File: src/statusline.ts
import { CommandType, SystemCommand } from "./cmd";
import getType from "./get-type";

export default function statusLine(data: SystemCommand, validInput: boolean = true): string {

    const name = data.username;
    if (!validInput) {
        return `Hey, you are wrong ${name}`;
    }

    const type = getType(data);

    if (type === CommandType.VimInsert || type === CommandType.VimAfter) {
        return `${name}: Has inserted ${data.message }`;
    }

    if (type === CommandType.VimCommand) {
        return `${name}: Vim Command ${data.message }`;
    }

    if (type === CommandType.SystemCommand) {
        return `${name}: ${data.message }`;
    }

    return `${name}: ${data.commandType} with ${data.message}`;
};

File: pkg/getStuff/statusLine.go
package getStuff

import (
	"fmt"

	"github.com/yonikosiner/twitch-bot/pkg/irc"
)

func StatusLine(data irc.IrcMessage) string {
	name := data.Name

	commandType := GetType(data)

	if commandType == irc.VimInsert || commandType == irc.VimAfter {
		return fmt.Sprintf("%s: Has inserted %s", name, data.Message)
	}

	if commandType == irc.VimCommand {
		return fmt.Sprintf("%s: Vim command %s", name, data.Message)
	}

	if commandType == irc.SystemCommand {
		return fmt.Sprintf("%s: %s", name, data.Message)
	}

	return fmt.Sprintf("%s: %d with %s", name, data.Type, data.Message)
}
